Projects/App/Sources/Screens/ProfileScreen/Screens/AboutScreen/AppInfo.swift

```diff
@@ -0,0 +1,222 @@
+                value: "© Hedvig AB - \(year)",
+                style: TextStyle.brand(.footnote(color: .primary)).centerAligned
+            )
+            footerView.textAlignment = .center
+
+            return state == .appInformation ? footerView : nil
+        }
+
+        form.appendSpacing(.inbetween)
+
+        let bodySection = form.appendSection(headerView: nil, footerView: footerView())
+
+        form.appendSpacing(.inbetween)
+
+        let buttonsSection = form.appendSection()
+
+        func value(row: State.InfoRows, completion: @escaping (String) -> Void) {
+            switch row {
+            case .language:
+                completion(Localization.Locale.currentLocale.displayName)
+            case .locale:
+                completion(Localization.Locale.currentLocale.market.marketName)
+            case .version:
+                completion(Bundle.main.appVersion)
+            case .memberId:
+                bag += client.fetch(query: GraphQL.MemberIdQuery()).valueSignal.compactMap {
+                    $0.member.id
+                }.onValue { memberId in
+                    completion(memberId)
+                }
+            }
+        }
+
+        func presentAlert() {
+            let alert = Alert(
+                title: L10n.settingsAlertChangeMarketTitle,
+                message: L10n.settingsAlertChangeMarketText,
+                tintColor: nil,
+                actions: [
+                    Alert.Action(
+                        title: L10n.alertOk,
+                        style: UIAlertAction.Style.destructive
+                    ) { true },
+                    Alert.Action(
+                        title: L10n.settingsAlertChangeMarketCancel,
+                        style: UIAlertAction.Style.cancel
+                    ) { false },
+                ]
+            )
+
+            bag += viewController.present(alert).onValue { shouldLogout in
+                if shouldLogout {
+                    ApplicationState.preserveState(.marketPicker)
+                    UIApplication.shared.appDelegate.logout()
+                }
+            }
+        }
+
+        func setupAppSettings() {
+            let row = State.InfoRows.locale
+            let market = Localization.Locale.currentLocale.market
+            bag += bodySection.append(AppInfoRow(title: row.title, icon: market.icon, isTappable: row.isTappable, value: market.marketName))
+
+            let language = State.InfoRows.language
+            let languageRow = AppInfoRow(title: language.title, icon: language.icon, isTappable: language.isTappable, value: Localization.Locale.currentLocale.displayName)
+            bag += bodySection.append(languageRow)
+
+            bag += languageRow.onSelect.onValue { _ in
+                presentAlert()
+            }
+        }
+
+        func setupAppInfo() {
+            [State.InfoRows.memberId, State.InfoRows.version].forEach { row in
+                value(row: row) { rowValue in
```

Comment:
Couldn't this lead to the ordering being non-deterministic? 😅

Reply:
Maybe we should add the row first and show a spinner until it has a value?

Reply:
@sampettersson trueeee